test/blas.jl

```diff
@@ -3,35 +3,36 @@
 import Base.LinAlg, Base.LinAlg.BlasReal, Base.LinAlg.BlasComplex
 
 srand(100)
-# syr2k! and her2k!
-for elty in (Float32, Float64, Complex64, Complex128)
-    U = randn(5,2)
-    V = randn(5,2)
-    if elty == Complex64 || elty == Complex128
-        U = complex.(U, U)
-        V = complex.(V, V)
+@testset "syr2k! and her2k!" begin
+    for elty in (Float32, Float64, Complex64, Complex128)
+        U = randn(5,2)
+        V = randn(5,2)
+        if elty == Complex64 || elty == Complex128
+            U = complex.(U, U)
+            V = complex.(V, V)
+        end
+        U = convert(Array{elty, 2}, U)
+        V = convert(Array{elty, 2}, V)
+        @test tril(LinAlg.BLAS.syr2k('L','N',U,V)) ≈ tril(U*V.' + V*U.')
+        @test triu(LinAlg.BLAS.syr2k('U','N',U,V)) ≈ triu(U*V.' + V*U.')
+        @test tril(LinAlg.BLAS.syr2k('L','T',U,V)) ≈ tril(U.'*V + V.'*U)
+        @test triu(LinAlg.BLAS.syr2k('U','T',U,V)) ≈ triu(U.'*V + V.'*U)
     end
-    U = convert(Array{elty, 2}, U)
-    V = convert(Array{elty, 2}, V)
-    @test tril(LinAlg.BLAS.syr2k('L','N',U,V)) ≈ tril(U*V.' + V*U.')
-    @test triu(LinAlg.BLAS.syr2k('U','N',U,V)) ≈ triu(U*V.' + V*U.')
-    @test tril(LinAlg.BLAS.syr2k('L','T',U,V)) ≈ tril(U.'*V + V.'*U)
-    @test triu(LinAlg.BLAS.syr2k('U','T',U,V)) ≈ triu(U.'*V + V.'*U)
-end
 
-for elty in (Complex64, Complex128)
-    U = randn(5,2)
-    V = randn(5,2)
-    if elty == Complex64 || elty == Complex128
-        U = complex.(U, U)
-        V = complex.(V, V)
+    for elty in (Complex64, Complex128)
+        U = randn(5,2)
+        V = randn(5,2)
+        if elty == Complex64 || elty == Complex128
+            U = complex.(U, U)
+            V = complex.(V, V)
+        end
+        U = convert(Array{elty, 2}, U)
+        V = convert(Array{elty, 2}, V)
+        @test tril(LinAlg.BLAS.her2k('L','N',U,V)) ≈ tril(U*V' + V*U')
+        @test triu(LinAlg.BLAS.her2k('U','N',U,V)) ≈ triu(U*V' + V*U')
+        @test tril(LinAlg.BLAS.her2k('L','C',U,V)) ≈ tril(U'*V + V'*U)
+        @test triu(LinAlg.BLAS.her2k('U','C',U,V)) ≈ triu(U'*V + V'*U)
     end
-    U = convert(Array{elty, 2}, U)
-    V = convert(Array{elty, 2}, V)
-    @test tril(LinAlg.BLAS.her2k('L','N',U,V)) ≈ tril(U*V' + V*U')
-    @test triu(LinAlg.BLAS.her2k('U','N',U,V)) ≈ triu(U*V' + V*U')
-    @test tril(LinAlg.BLAS.her2k('L','C',U,V)) ≈ tril(U'*V + V'*U)
-    @test triu(LinAlg.BLAS.her2k('U','C',U,V)) ≈ triu(U'*V + V'*U)
 end
 
 ## BLAS tests - testing the interface code to BLAS routines
```

Comment:
Making this `for` loop a `@testset` and putting the `elty` in the test string will make it easier to locate the error.